compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7606,6 +7608,21 @@ private BType checkMappingField(RecordLiteralNode.RecordField field, BType mappi
                             }
                         }
                     }
+                    if (!spreadRecordType.sealed) {
```

Comment:
With this change, for the following example we are getting different error compared to the previous one.
```ballerina
type Address record {
    string street;
};

public function main() {
    Address address = {street: "Main Street", "number": 40};
    record {|string street;|} _ = {street: "Main Street", ...address};
}
```

In version 2201.8.0, the error message was 'invalid usage of record literal: duplicate key 'street' via spread operator '...address'. Now, with the recent changes, we are encountering a different error: 'invalid usage of spread field with open record of type 'Address' that may have rest fields to construct a closed record'